bool release
bool softMode
uint8 forceLevel
bool gripInside
float64 currentStrokeMM